internal/service/appstream/stack.go

```diff
@@ -195,6 +196,12 @@ func ResourceStack() *schema.Resource {
 							Required:         true,
 							ValidateDiagFunc: enum.Validate[awstypes.Action](),
 						},
+						"maximum_length": {
+							Type:         nullable.TypeNullableInt,
+							Optional:     true,
+							Default:      "",
+							ValidateFunc: nullable.ValidateTypeStringNullableIntBetween(1, 20971520),
```

Comment:
I've got some doubt that I approached this schema addition correctly. It appears to work as expected (acceptance tests pass) but I'd appreciate advice on how to approach an optional integer. Here's an example of how it looks in the Terraform state

```
           "user_settings": [
              {
                "action": "CLIPBOARD_COPY_FROM_LOCAL_DEVICE",
                "maximum_length": "10000",
                "permission": "ENABLED"
              },
              {
                "action": "CLIPBOARD_COPY_TO_LOCAL_DEVICE",
                "maximum_length": "1000000",
                "action": "CLIPBOARD_COPY_TO_LOCAL_DEVICE",
                "maximum_length": "1000000",
                "permission": "ENABLED"
              },
              {
                "action": "DOMAIN_PASSWORD_SIGNIN",
                "maximum_length": "",
                "permission": "ENABLED"
              },
              {
                "action": "DOMAIN_SMART_CARD_SIGNIN",
                "maximum_length": "",
                "permission": "DISABLED"
              },
              {
                "action": "FILE_DOWNLOAD",
                "maximum_length": "",
                "permission": "DISABLED"
              },
              {
                "action": "FILE_UPLOAD",
                "maximum_length": "",
                "permission": "ENABLED"
              },
              {
                "action": "PRINTING_TO_LOCAL_DEVICE",
                "maximum_length": "",
                "permission": "DISABLED"
              }
            ]
```